### PR TITLE
Get Account Asset transactions pagination

### PR DESCRIPTION
**Pull request by nickaleks**
### Description of the Change

This PR introduces pagination logic to GetAccountAssetTransactions query.

### Benefits
Pagination for query

### Possible Drawbacks 

Since logic of getAccountTransactions and GetAccountAssetTransactions is almost the same, their logic was moved to a common function. This function might be confusing.

### CODE REVIEW

File: irohad/ametsuchi/impl/postgres_query_executor.cpp [hunk 1]
```diff
@@ -342,6 +342,119 @@ namespace iroha {
...
+            position_by_hash.index >= first_hash.index)
+      ),
+      total_size AS (
+        SELECT COUNT(*) FROM my_txs
+      ),
+      t AS (
+        SELECT my_txs.height, my_txs.index
+        FROM my_txs
+        JOIN previous_txes ON my_txs.height = previous_txes.height
+        AND my_txs.index = previous_txes.index
```
Comment (MBoldyrev) [inline irohad/ametsuchi/impl/postgres_query_executor.cpp]: And do we need that additional temporary table at all? I mean current `previous_txes`. Seems like we can write
```sql
        JOIN first_hash ON my_txs.height > first_hash.height
        OR (my_txs.height = first_hash.height AND
            my_txs.index >= first_hash.index)
```
and remove the `previous_txes` completely

File: irohad/ametsuchi/impl/postgres_query_executor.cpp [hunk 2]
```diff
@@ -342,6 +342,119 @@ namespace iroha {
...
+        LIMIT :page_size
+      )
+      SELECT height, index, count, perm FROM t
+      RIGHT OUTER JOIN has_perms ON TRUE
```
Comment (MBoldyrev) [inline irohad/ametsuchi/impl/postgres_query_executor.cpp]: Quite strange join. Maybe place a todo to rewrite it?

Comment (nickaleks) [reply irohad/ametsuchi/impl/postgres_query_executor.cpp]: We want to add permissions to the result of the query, that's why we join

Comment (MBoldyrev) [reply irohad/ametsuchi/impl/postgres_query_executor.cpp]: It is strange that we use `RIGHT OUTER JOIN .. ON TRUE`, while it seems we could use just `CROSS JOIN`

**Pull request merged**